### PR TITLE
API 2.0 schema re-work

### PR DESCRIPTION
**Pull request by troeger**
This is an attempted fix for #135, which introduces unique keys in all dictionaries used the API. The intention here is to simplify the mapping to input forms in the new frontend.

### CODE REVIEW

**Review by DenizCankurtaran — COMMENTED** [review]
It feels much cleaner, thank you!

File: kubeportal/k8s/kubernetes_api.py [hunk 1]
```diff
@@ -75,14 +75,16 @@ def create_k8s_deployment(namespace: str, name: str, replicas: int, match_labels
     """
     logger.info(f"Creating Kubernetes deployment '{name}'")
     k8s_containers = [client.V1Container(name=c["name"], image=c["image"]) for c in tpl["containers"]]
+    k8s_labels = {item['key']:item['value'] for item in tpl['labels']}
     k8s_pod_tpl = client.V1PodTemplateSpec(
-        metadata=client.V1ObjectMeta(name=tpl['name'], labels=tpl['labels']),
+        metadata=client.V1ObjectMeta(name=tpl['name'], labels=k8s_labels),
```
Comment (kat-hi) [inline kubeportal/k8s/kubernetes_api.py:80]: pls rnm tpl

**Review by kat-hi — CHANGES_REQUESTED** [review]

File: kubeportal/k8s/kubernetes_api.py (hunk 1 above)
Comment (troeger) [reply kubeportal/k8s/kubernetes_api.py]: fxd

Comment (DenizCankurtaran) [reply kubeportal/k8s/kubernetes_api.py]: btfl

**Pull request merged**